src/input-number/input-number.tsx

```diff
@@ -0,0 +1,47 @@
+import clsx from "clsx";
+import { Heading } from "../heading";
+import { Text } from "../text";
+import React, { useState } from "react";
+
+type Props = {
+  values: { value: string; label: string | React.ReactNode }[];
+  value?: string | undefined;
+  onClick?: (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+};
+
+export const InputNumber = ({ values, value, onClick }: Props) => {
+  const [selected, setSelected] = useState(value);
+  const valuesById = Object.fromEntries(
+    values.map((value) => [value.value, value.label])
+  );
+  const handleOnClick = (e: React.MouseEvent<HTMLDivElement>) => {
+    setSelected(e.target.textContent);
```

Comment:
why are you keeping a copy of the local state? you don't need it, use the value you get as prop